kafka/src/kafka/impl/consumer_impl.cpp

```diff
@@ -292,6 +294,53 @@ void ConsumerImpl::Commit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*
 
 void ConsumerImpl::AsyncCommit() { rd_kafka_commit(consumer_.GetHandle(), nullptr, /*async=*/1); }
 
+std::pair<std::int64_t, std::int64_t> ConsumerImpl::GetMinMaxOffset(const std::string& topic, std::int32_t partition) {
+    std::int64_t offset_high{-1};
+    std::int64_t offset_low{-1};
+
+    auto err = rd_kafka_query_watermark_offsets(
+        consumer_.GetHandle(), topic.data(), partition, &offset_low, &offset_high, /*timeout_ms=*/-1
+    );
+    if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+        LOG_ERROR() << fmt::format("Failed to get offsets: {}", rd_kafka_err2str(err));
+        return {-1, -1};
+    }
+
+    if (offset_high == RD_KAFKA_OFFSET_INVALID) {
+        LOG_ERROR() << fmt::format("Invalid offset for topic {} partition {}", topic, partition);
+        return {-1, -1};
+    }
+
+    return {offset_low, offset_high};
+}
+
+std::vector<std::int32_t> ConsumerImpl::GetPartitionsId(const std::string& topic) {
+    const rd_kafka_metadata_t* raw_metadata{nullptr};
+    auto err = rd_kafka_metadata(consumer_.GetHandle(), 0, nullptr, &raw_metadata, /*timeout_ms=*/-1);
+    if (err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+        LOG_ERROR() << fmt::format("Failed to fetch metadata: ", rd_kafka_err2str(err));
+        return {};
+    }
+    MetadataHolder metadata{raw_metadata};
+
+    auto topics = boost::make_iterator_range(metadata->topics, metadata->topics + metadata->topic_cnt);
+    auto* topic_it = std::find_if(topics.begin(), topics.end(), [&topic](const rd_kafka_metadata_topic& topic_raw) {
+        return topic == topic_raw.topic;
+    });
+    if (topic_it == topics.end()) {
+        LOG_ERROR() << fmt::format("Topic not found: {}", topic);
+        return {};
+    }
+
+    std::vector<std::int32_t> partitions_id;
```

Comment:
use `std::vector::reserve`, because you know the size

Reply:
Done